src/azure-cli/azure/cli/command_modules/cdn/custom.py

```diff
@@ -71,6 +91,304 @@ def update_endpoint(instance,
+            ))
+    if match_variable == 'Cookies':
+        return DeliveryRuleCookiesCondition(
+            parameters=CookiesMatchConditionParameters(
+                operator=operator,
+                selector=selector,
+                match_values=match_values.split(","),
+                negate_condition=negate_condition,
+                transforms=transform
+            ))
+    return None
+
+
+# pylint: disable=too-many-return-statements
+def create_action(action_name, cache_behavior=None, cache_duration=None, header_action=None,
+                  header_name=None, header_value=None, query_string_behavior=None, query_parameters=None,
+                  redirect_type=None, redirect_protocol=None, custom_hostname=None, custom_path=None,
+                  custom_query_string=None, custom_fragment=None, source_pattern=None, destination=None,
+                  preserve_unmatched_path=None):
+    if action_name == "CacheExpiration":
+        return DeliveryRuleCacheExpirationAction(
+            parameters=CacheExpirationActionParameters(
+                cache_behavior=cache_behavior,
+                cache_duration=cache_duration
+            ))
+    if action_name == 'RequestHeader':
+        return DeliveryRuleRequestHeaderAction(
+            parameters=HeaderActionParameters(
+                header_action=header_action,
+                header_name=header_name,
+                value=header_value
+            ))
+    if action_name == 'ResponseHeader':
+        return DeliveryRuleResponseHeaderAction(
+            parameters=HeaderActionParameters(
+                header_action=header_action,
+                header_name=header_name,
+                value=header_value
+            ))
+    if action_name == "CacheKeyQueryString":
+        return DeliveryRuleCacheKeyQueryStringAction(
+            parameters=CacheKeyQueryStringActionParameters(
+                query_string_behavior=query_string_behavior,
+                query_parameters=query_parameters
+            ))
+    if action_name == 'UrlRedirect':
+        return UrlRedirectAction(
+            parameters=UrlRedirectActionParameters(
+                redirect_type=redirect_type,
+                destination_protocol=redirect_protocol,
+                custom_path=custom_path,
+                custom_hostname=custom_hostname,
+                custom_query_string=custom_query_string,
+                custom_fragment=custom_fragment
+            ))
+    if action_name == 'UrlRewrite':
+        return UrlRewriteAction(
+            parameters=UrlRewriteActionParameters(
+                source_pattern=source_pattern,
+                destination=destination,
+                preserve_unmatched_path=preserve_unmatched_path
+            ))
+    return DeliveryRuleAction()
+
+
+# pylint: disable=too-many-locals
+def add_rule(client, resource_group_name, profile_name, endpoint_name,
+             order, rule_name, action_name, match_variable=None, operator=None,
```

Comment:
Are all of these arguments optional? Can user create an empty rule with a name only?

Reply:
can we add an empty rule?

Reply:
There parameters is required by our API and cannot be empty. Empty rule is not allowed.